ocaml/xapi/xapi_vm_helpers.ml

```diff
@@ -154,18 +151,13 @@ let update_vm_virtual_hardware_platform_version ~__context ~vm =
 
 let create_from_record_without_checking_licence_feature_for_vendor_device
     ~__context rpc session_id vm_record =
+  (* vendor device is no longer licensed *)
   let mk_vm r =
     Client.Client.VM.create_from_record ~rpc ~session_id
       ~value:{r with API.vM_suspend_VDI= Ref.null; API.vM_power_state= `Halted}
   in
   let has_vendor_device = vm_record.API.vM_has_vendor_device in
-  if
-    has_vendor_device
-    && not
-         (Pool_features.is_enabled ~__context
-            Features.PCI_device_for_auto_update
-         )
-  then (
+  if has_vendor_device then (
     (* Avoid the licence feature check which is enforced in VM.create (and create_from_record). *)
     let vm = mk_vm {vm_record with API.vM_has_vendor_device= false} in
     Db.VM.set_has_vendor_device ~__context ~self:vm ~value:true ;
```

Comment:
When the license checking is removed, the settings on `has_vendor_device` (to `false` and then to `true`) are not necessary. So they can be removed.
But I can't see the relationship between the feature and `hardware_platform_version`.